lib/elixir/lib/kernel.ex

```diff
@@ -3870,24 +3877,28 @@ defmodule Kernel do
       end
 
   However, do not provide `__using__/1` if all it does is to import,
-  alias or require the module itself. For example, do not:
+  alias or require the module itself. For example, avoid this:
 
       defmodule MyModule do
-        defmacro __using__(opts) do
+        defmacro __using__(_opts) do
           quote do
             import MyModule
           end
         end
       end
 
-  In such cases, developers must just import or alias the module
-  directly, allowing developers to customize those as they wish,
+  In such cases, developers should just import or alias the module
+  directly, so that they can customize those as they wish,
   without the indirection behind `use/2`.
 
   Finally, developers should also avoid defining functions inside
   the `__using__/1` callback, unless those functions are the default
-  implementation of a previously defined `@callback`. In case you
-  want to provide some existing functionality to the user module,
+  implementation of a previously defined `@callback` or are functions
+  meant to be overridden (see `defoverridable/1`). Even in these cases,
+  defining functions should be seen as a "last resource" because of the
+  complexity that "magically" defined function can introduce.
+
+  In case you want to provide some existing functionality to the user module,
```

Comment:
Maybe we should expand on this a bit? For example, in ExUnit.Case, we don't define test in the user module when you  `use ExUnit`, instead it invokes the "test/2" macro defined by ExUnit.Case and imported on use.